packages/embark/src/cmd/cmd_controller.js

```diff
@@ -352,6 +352,43 @@ class EmbarkController {
     });
   }
 
+  exec(options, callback) {
+
+    const engine = new Engine({
+      env: options.env,
+      embarkConfig: options.embarkConfig || 'embark.json'
+    });
+
+    engine.init({}, () => {
+      engine.registerModuleGroup("coreComponents", {
+        disableServiceMonitor: true
+      });
+      engine.registerModuleGroup("stackComponents");
+      engine.registerModuleGroup("blockchain");
+      engine.registerModuleGroup("compiler");
+      engine.registerModuleGroup("contracts");
+      engine.registerModulePackage('embark-deploy-tracker', {
+        plugins: engine.plugins
+      });
+      engine.registerModulePackage('embark-scripts-runner', engine.plugins);
```

Comment:
yeah this should be an options object indeed.
This sort of options are on the way out as well, I don't recall though if the changes that make it unnecesary are already in or in another branch (i.e one can use or will be able to use `embark.plugins`)

Reply:
One of the reason I went with passing plugins directly is because it is **not** an option. The plugin wouldn't work without this dependency. 

> This sort of options are on the way out as well, 

That's good to know! Options should be options, not dependencies.

Reply:
Changed it to options for consistency. Still think that options are a bad way to represent something that is a required dependency